controller/workitemtype.go

```diff
@@ -101,16 +101,26 @@ func ConvertWorkItemTypeFromModel(request *http.Request, t *workitem.WorkItemTyp
 	return converted
 }
 
-// converts the field type from modesl to app representation
+// converts the field type from model to app representation
 func ConvertFieldTypeFromModel(t workitem.FieldType) app.FieldType {
 	result := app.FieldType{}
 	result.Kind = string(t.GetKind())
-	switch t2 := t.(type) {
+	switch modelFieldType := t.(type) {
 	case workitem.ListType:
-		result.ComponentType = ptr.String(string(t2.ComponentType.GetKind()))
+		result.ComponentType = ptr.String(string(modelFieldType.ComponentType.GetKind()))
+		if modelFieldType.DefaultValue != nil {
+			result.DefaultValue = &modelFieldType.DefaultValue
+		}
 	case workitem.EnumType:
-		result.BaseType = ptr.String(string(t2.BaseType.GetKind()))
-		result.Values = t2.Values
+		result.BaseType = ptr.String(string(modelFieldType.BaseType.GetKind()))
+		result.Values = modelFieldType.Values
+		if modelFieldType.DefaultValue != nil {
+			result.DefaultValue = &modelFieldType.DefaultValue
+		}
+	case workitem.SimpleType:
+		if modelFieldType.DefaultValue != nil {
```

Comment:
No, the `modelFieldType.DefaultValue` is not the same for all cases in the switch. The switch is a type assertion switch `switch modelFieldType := t.(type)` and therefore `modelFieldType` is always something else. Please note, that I intentionally define a default value on each of the types itself:

```go
type SimpleType struct {
	Kind         Kind        `json:"kind"`
	DefaultValue interface{} `json:"default_value,omitempty"`
}

type EnumType struct {
	SimpleType       `json:"simple_type"`
	BaseType         SimpleType    `json:"base_type"`
	Values           []interface{} `json:"values"`
	RewritableValues bool          `json:"rewritable_values"`
	DefaultValue     interface{}   `json:"default_value,omitempty"`
}

type ListType struct {
	SimpleType    `json:"simple_type"`
	ComponentType SimpleType  `json:"component_type"`
	DefaultValue  interface{} `json:"default_value,omitempty"`
}
```

I cannot re-use the `DefaultValue` defined from `SimpleType` in `EnumType` or `ListType` because the default value would then be completely unaware of the allowed values in an enum for example.

Reply:
I had a discussion with @kwk and we agreed that this is the appropriate way to do it.